src/osd/PG.cc

```diff
@@ -5944,7 +5944,7 @@ boost::statechart::result PG::RecoveryState::Peering::react(const AdvMap& advmap
     post_event(advmap);
     return transit< Reset >();
   }
-  
+  pg->check_full_transition(advmap.lastmap, advmap.osdmap);
```

Comment:
Oops, that's true and I am going to drop this change now. And thanks for the review:)